framework/src/engine/consensus/certificate_generation/commit_pool.ts

```diff
@@ -210,11 +212,13 @@ export class CommitPool {
 		}
 
 		try {
-			const heightNextBFTParameters = await this._bftMethod.getNextHeightBFTParameters(
+			let heightNextBFTParameters = await this._bftMethod.getNextHeightBFTParameters(
 				stateStore,
 				maxHeightCertified + 1,
 			);
 
+			heightNextBFTParameters = Math.max(heightNextBFTParameters, this._minCertifyHeight);
```

Comment:
Similarly to my comment in `selectAggregateCommit`, I would revert this change here and just modify the check of line 206 

`if aggregateCommit.height <= Math.max(maxHeightCertified, minCertifyHeight -1)`